src/components/PeerAddressHintTextField.tsx

```diff
@@ -0,0 +1,225 @@
+import { Tooltip } from "@material-ui/core";
+import { InputBaseComponentProps } from "@material-ui/core/InputBase";
+import Paper from "@material-ui/core/Paper";
+import {
+  createStyles,
+  makeStyles,
+  Theme,
+  useTheme
+} from "@material-ui/core/styles";
+import { TextFieldProps } from "@material-ui/core/TextField";
+import React, { CSSProperties, HTMLAttributes } from "react";
+import Select from "react-select";
+import {
+  ValueContainer as RSValueContainer,
+  ValueContainerProps
+} from "react-select/lib/components/containers";
+import { ControlProps } from "react-select/lib/components/Control";
+import { MenuProps } from "react-select/lib/components/Menu";
+import { ValueType } from "react-select/lib/types";
+import TextField from "./TextField";
+
+interface PeerAddressHintTextFieldProps {
+  addressHint: string;
+  onAddressHintChange?: (newAddressHint: string) => void;
+  disabled?: boolean;
+  suggestions?: string[];
+}
+
+interface OptionType {
+  label: string;
+  value: string;
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    input: {
+      display: "flex",
+      padding: 0,
+      height: "auto"
+    },
+    valueContainer: {
+      // The `emotion` styles of react-select are injected _after_ our styling, hence we need to workaround with !important.
+      // If we can find out, how to change the injection position of `emotion`'s <style> tags, we can fix this.
+      padding: "0 !important"
+    },
+    paper: {
+      position: "absolute",
+      zIndex: 1,
+      marginTop: theme.spacing(1),
+      left: 0,
+      right: 0
+    }
+  })
+);
+
+interface InnerTextFieldProps {
+  addressHint: string;
+  onAddressHintChange: (newAddressHint: string) => void;
+  disabled: boolean;
+}
+
+function InnerTextField({
+  addressHint,
+  onAddressHintChange,
+  disabled,
+  ...baseTextFieldProps
+}: InnerTextFieldProps & TextFieldProps) {
+  return (
+    <TextField
+      value={addressHint}
+      label={"Peer Address Hint"}
+      helperText={
+        <Tooltip
+          title={
+            // Could make this sentence less awkward if we pass in, whether or not the PeerID belongs to our COMIT node...
+            "Supply a publicly reachable address here if the COMIT node with the above PeerID is not discoverable."
```

Comment:
```suggestion
            "Supply a multiaddress hint here if the COMIT node with the provided PeerID cannot be discovered."
```

🙃 not sure my sentence is much better, maybe it'll inspire you.